Add verify-golint.sh

cici37 opened the pull request:
Added verify-golint.sh. Updated vendor accordingly. Fixed some golint failures

cici37 commented:
/assign @jpbetz

jpbetz commented:
the go.mod change looks fine (mostly labeling things as indirect and adding the linter dep), but do we need all the vendor changes? I was only expecting linter related dependency changes in vendor? This seems to have pulled in quite a bit more.

cici37 commented:
> the go.mod change looks fine (mostly labeling things as indirect and adding the linter dep), but do we need all the vendor changes? I was only expecting linter related dependency changes in vendor? This seems to have pulled in quite a bit more.

etcd related dependency changes is due to previous update.  
Changes under golang.org/x is due to go.mod updates. I re-run `go mod tidy` to make go.mod more clear now.

jpbetz commented:
/lgtm

cici37 commented:
/assign @bowei

bowei commented on tools/verify-golint.sh:
```diff
@@ -0,0 +1,159 @@
+# Ensure that we find the binaries we build before anything else.
+KUBE_ROOT_ABSOLATE="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
+
+export GOBIN="${KUBE_ROOT_ABSOLATE}/tools"
```
typo? ABSOLATE => ABSOLUTE

cici37 replied:
Updated

bowei commented on tools/verify-golint.sh:
```diff
@@ -0,0 +1,159 @@
+#kube::golang::verify_go_version
+
+# Ensure that we find the binaries we build before anything else.
+KUBE_ROOT_ABSOLATE="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd -P)"
```
typo? ABSOLATE => ABSOLUTE

bowei commented on tools/verify-golint.sh:
```diff
@@ -0,0 +1,159 @@
+# Install golint
+echo 'installing golint'
+pushd "${KUBE_ROOT}/tools" >/dev/null
+  GO111MODULE=on go get -u golang.org/x/lint/golint
```
Shouldn't we run this stuff in a container?

bowei commented on tools/verify-golint.sh:
```diff
@@ -0,0 +1,159 @@
+
+# arguments: target, item1, item2, item3, ...
+# returns 0 if target is in the given items, 1 otherwise.
+array_contains() {
```
Is this a utility that should be put into a common library?

cici37 replied:
Added a lib dir and moved to util.sh

bowei commented on tools/verify-golint.sh:
```diff
@@ -0,0 +1,159 @@
+
+# arguments: target, item1, item2, item3, ...
+# returns 0 if target is in the given items, 1 otherwise.
+array_contains() {
```
[minor] you use `function xyz { }` above and `xyz() { }` syntax, can we make it consistent

bowei commented on tools/verify-golint.sh:
```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env bash
```
Is this copied from somewhere else? Which pieces have changed?

cici37 replied:
Updated the commit message

bowei commented:
Can you squash the two vendor/ changes together?
Can you add information in your commit message about where the go lint sh script comes from?

cici37 commented on tools/verify-golint.sh:
```diff
@@ -0,0 +1,159 @@
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+#source "${KUBE_ROOT}/hack/lib/init.sh"
```
Thanks for reviewing. This is outdated and has been removed before.

cici37 commented:
> Can you squash the two vendor/ changes together?
> Can you add information in your commit message about where the go lint sh script comes from?

Hi @bowei Thank you for the review. I have squashed the two update vendor commit and addressed other review comments in the latest commit. Could you please take a look when you have time? Thank you

bowei commented:
/lgtm
/approve
